### PR TITLE
Upgrade Microsoft.ApplicationInsights from 2.4.0 to 2.12.0

### PR DESCRIPTION
**Pull request by msJinLei**
- Upgrade Microsoft.ApplicationInsights from 2.4.0 to 2.12.0

### CODE REVIEW

File: src/Common/MetricHelper.cs [hunk 1]
```diff
@@ -425,7 +425,7 @@ private void PopulatePropertiesFromQos(AzurePSQoSEvent qos, IDictionary<string,
         private static string[] exceptionTrackAcceptModuleList = { "Az.Accounts", "Az.Compute", "Az.AKS", "Az.ContainerRegistry" };
         private static string[] exceptionTrackAcceptCmdletList = { "Get-AzKeyVaultSecret", "Get-AzKeyVaultCert" };
 
-        private static string ConvertFrameToString(StackFrame frame)
+        private static string ConvertFrameToString(System.Diagnostics.StackFrame frame)
```
Comment (isra-fel) [inline src/Common/MetricHelper.cs:428]: Change on this line doesn't seem necessary

File: src/Common/MetricHelper.cs [hunk 2]
```diff
@@ -127,7 +127,7 @@ public MetricHelper(INetworkHelper network)
 #if DEBUG
             if (TestMockSupport.RunningMocked)
             {
-                TelemetryConfiguration.Active.DisableTelemetry = true;
+                telemetryConfiguration.DisableTelemetry = true;
```
Comment (isra-fel) [inline src/Common/MetricHelper.cs:130]: The active telemetry configuration being used may not be `telemetryConfiguration`, so I suppose `TelemetryConfiguration.Active` makes more sense here

**Review by isra-fel — CHANGES_REQUESTED** [review]
~~Have you tested it out with powershell modules?~~
Jin Lei has tested locally and judged from the traces the telemetry data was sent to Azure

File: src/Common/MetricHelper.cs (hunk 1 above)
Comment (msJinLei) [reply src/Common/MetricHelper.cs:428]: StackFrame also defined in new version of Application Insights namespace and so I add prefix here.

File: src/Common/MetricHelper.cs (hunk 2 above)
Comment (msJinLei) [reply src/Common/MetricHelper.cs:130]: Using TelemetryConfiguration.Active is not recommended on .NET Core. See https://github.com/microsoft/ApplicationInsights-dotnet/issues/1152 for more details

**Review by isra-fel — APPROVED** [review]

**Pull request merged**